packages/flash-swap/test/shared/fixtures.ts

```diff
@@ -70,15 +72,23 @@ export async function integrationFixture(signers: Signer[]): Promise<Integration
     ])
   );
 
-  const godModeUniswapV2PairArtifact: Artifact = await artifacts.readArtifact("GodModeUniswapV2Pair");
-  const uniswapV2Pair: GodModeUniswapV2Pair = <GodModeUniswapV2Pair>(
-    await deployContract(deployer, godModeUniswapV2PairArtifact, [])
+  const uniswapV2FactoryArtifact: Artifact = await artifacts.readArtifact("UniswapV2Factory");
+  const uniswapV2Factory: UniswapV2Factory = <UniswapV2Factory>(
+    await deployContract(deployer, uniswapV2FactoryArtifact, [await deployer.getAddress()])
   );
-  await uniswapV2Pair.initialize(wbtc.address, usdc.address);
+  await uniswapV2Factory.createPair(usdc.address, wbtc.address);
+  const pairAddress = await uniswapV2Factory.allPairs(0);
+
+  const uniswapV2PairArtifact: Artifact = await artifacts.readArtifact("UniswapV2Pair");
+  const uniswapV2Pair = new Contract(pairAddress, uniswapV2PairArtifact.abi, deployer) as UniswapV2Pair;
```

Comment:
Instead of `new Contract` we should import the `UniswapV2Pair__factory` object generated by TypeChain and use the `connect` method .